plugins/importexport/native/filter/NativeXmlRepresentationFilter.php

```diff
@@ -113,9 +115,24 @@ public function parseIdentifier($element, $representation)
                 break;
             default:
                 if ($advice == 'update') {
-                    // Load pub id plugins
-                    $pubIdPlugins = PluginRegistry::loadCategory('pubIds', true, $deployment->getContext()->getId());
-                    $representation->setStoredPubId($element->getAttribute('type'), $element->textContent);
+                    if ($element->getAttribute('type') == 'doi') {
+                        if ($doiObject = $representation->getData('doiObject')) {
+                            Repo::doi()->edit($doiObject, ['doi' => $element->textContent]);
```

Comment:
This might be a larger problem beyond the scope of this PR, but since the DOIs are either being directly edited or created here, if something fails later on, this data will already be written to the database. I don't know if there is a good solution currently for the within the Native XML toolset, but thought I would flag it.

Reply:
When there already is a `$doiObject` present, I wonder if we should also update the `status` to `Doi::STATUS_STALE` since the metadata is changing and will need updating with whichever registration agency is being used?

Reply:
Hmmmm.... Now that I think more about it: actually the object from any`NativeXml...Filter` will never already have a DOI, because this object has just been newly created and imported. The update of objects by using native import is currently not supported, s. e.g. https://github.com/pkp/pkp-lib/blob/main/plugins/importexport/native/filter/NativeXmlRepresentationFilter.php#L106.
Thus, I would suggest to remove that `if ($doiObject = $representation->getData('doiObject'))` completely. OK?

The other thing is to ensure that the DOI from the import XML file does not already exist (which is automatically done within `Repo::doi()->add()`, right?)  i.e. if it exist to use its ID. 
Thus in the code next/below I would first check if such a DOI exists (`filterByIdentifier()`?). 
If it exists the object will get its ID. This could, for example, be the case for all publications of an article. I am not sure if I could also check that the existing DOI is assigned to an object of the same type? :thinking: Else, we could for now just assume that the XML file is correct.
If it does not exist, a new DOI will be created and inserted, as in the code next below.

And, yes, if then a new DOI is created and something goes wrong, so that the object is not updated in the DB, then the DOI would not be linked to anything :-P I think it is similar with other metadata, e.g. keywords, subjects, or even authors... :thinking: ...so that I do not know the solution for that now...

Reply:
> Thus, I would suggest to remove that if ($doiObject = $representation->getData('doiObject')) completely. OK?

That sounds good to me.

> The other thing is to ensure that the DOI from the import XML file does not already exist (which is automatically done within Repo::doi()->add(), right?)

I took a look and this is not done automatically as part of `Repo::doi()->add()`. It's instead handled as part of `Repo::doi()->validate()`. Here's the code that does the check, though it may be worth it running the `validate` method before saving anyway:

```php
// Check for duplicates across all contexts
$doiId = $object ? $object->getData('id') : null;
$doi = $props['doi'] ?? null;
if ($doi !== null && $this->isDuplicate($doi, $doiId)) {
    $validator->errors()->add('doi', __('doi.editor.doiSuffixCustomIdentifierNotUnique'));
}
```

`filterByIdentifier()` would do the check and is what's used within `isDuplicate()`.

> I think it is similar with other metadata, e.g. keywords, subjects, or even authors... 🤔 ...so that I do not know the solution for that now...

That's what I assumed, but thought I'd check.